src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,362 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { base } from '$app/paths';
+  import type { ICellRendererParams } from 'ag-grid-community';
+  import Truck from 'bootstrap-icons/icons/truck.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import {
+    derivationGroupPlanLinkError,
+    derivationGroups,
+    externalSources,
+    selectedPlanDerivationGroupNames,
+  } from '../../stores/external-source';
+  import { plan } from '../../stores/plan';
+  import { plugins } from '../../stores/plugins';
+  import { view } from '../../stores/views';
+  import type { User } from '../../types/app';
+  import type { DataGridColumnDef } from '../../types/data-grid';
+  import type { DerivationGroup, ExternalSourceSlim } from '../../types/external-source';
+  import type { Layer } from '../../types/timeline';
+  import effects from '../../utilities/effects';
+  import { formatDate } from '../../utilities/time';
+  import { isExternalEventLayer } from '../../utilities/timeline';
+  import { showFailureToast } from '../../utilities/toast';
+  import Collapse from '../Collapse.svelte';
+  import Input from '../form/Input.svelte';
+  import CssGrid from '../ui/CssGrid.svelte';
+  import CssGridGutter from '../ui/CssGridGutter.svelte';
+  import DataGrid from '../ui/DataGrid/DataGrid.svelte';
+  import DataGridActions from '../ui/DataGrid/DataGridActions.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import SectionTitle from '../ui/SectionTitle.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let user: User | null;
+
+  type CellRendererParams = {
+    viewDerivationGroup: (derivationGroup: DerivationGroup) => void;
+  };
+  type DerivationGroupCellRendererParams = ICellRendererParams<DerivationGroup> & CellRendererParams;
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+  }>();
+
+  let externalEventLayers: Layer[] | undefined;
+  let dataGrid: DataGrid<DerivationGroup>;
+  let baseColumnDefs: DataGridColumnDef<DerivationGroup>[] = [];
+
+  selectedPlanDerivationGroupNames.subscribe(() => {
+    if (baseColumnDefs.length > 0 && dataGrid) {
```

Comment:
I don't think that `baseColumnDefs` needs to be a dependency here.

Reply:
It was placed there originally as `baseColumnDefs` was not set at initialization as that wasn't playing well with the involved stores, and we didn't want to invoke this refresh before the column definitions were properly initialized. However, on refactoring `baseColumnDefs` to merge with `columnDefs`, and either replacing this line with a check on `columnDefs.length` or removing it altogether, we found that both worked. To reduce dependencies, we will remove it outright per your suggestion.

Reply:
See [this commit](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/5756e9fe9c0f49dd6f0454df66f03882053a7f0a).